LayoutTests/http/tests/inspector/console/message-from-worker.html

```diff
@@ -0,0 +1,61 @@
+<script src="/inspector/resources/inspector-test.js"></script>
+<script>
+    function test() {
+        // The purpose of these tests:
+        // Web Inspector should receive console messages generated from workers correctly.
+
+        const suite = InspectorTest.createAsyncSuite("message-from-worker");
+
+        // Utilities for test code.
+        let callback = () => {};
+        WI.consoleManager.addEventListener(WI.ConsoleManager.Event.MessageAdded, (event) => {
+            InspectorTest.log(`Got message: ${event.data.message._messageText}`);
+            callback();
+        });
+
+        function runMessageTest(resolve, expected, func) {
+            let seen = 0;
+
+            callback = () => {
+                if (++seen === expected)
+                    resolve();
+            };
+
+            InspectorTest.evaluateInPage(func);
+        }
+        // Utilities for page code.
+        InspectorTest.evaluateInPage(`
+            function addIFrame(url, testId) {
+                const iframe = document.createElement("iframe");
+                iframe.src = url;
+                iframe.onload = () => console.log("iframe is loaded in " + location.href + " [" + testId + "]");
+                document.body.appendChild(iframe);
+            }
+        `);
```

Comment:
I believe you can actually declare this `addIFrame` function in the test's script without using `evaluateInPage`, by moving this declaration to before `function test()` near the top. It's probably more readable that way, if that works.

Reply:
Oh, you mean before test()?